src/symbolicate_response_assembler.rs

```diff
@@ -0,0 +1,235 @@
+use super::*;
+
+/**
+ * This module is for gluing `mod symbolicate` and other modules (such as `compact_symbol_table`)
+ * or Javascript callbacks together to assemble a symbolication response from a json request.
+ */
+
+/// Receives a job and then returns a response stack.
+/// * `symbolicate_job`: The parsed symbolicateJob from the json request
+/// * `module_index`: index of the module in the stacks of the symbolicate request
+/// * `candidate_paths`: candidate paths specific to this module
+/// * `read_buffer_callback` test a pair of candidate paths (binary and debug) for a module
+///     retrieve the module in the form of binary buffer if valid, else return Err()
+///
+/// To assemble the resulting stack into a symbolicate job response, caller will
+/// need to read the information from the stack (such as function module)
+/// and insert this the stack into the response.
+pub async fn assemble_response_stack<'a>(
+    symbolicate_job: &'a SymbolicateJob,
+    module_index: usize,
+    candidate_paths: Vec<CandidatePath>,
+    read_buffer_callback: &'a js_sys::Function,
```

Comment:
I think it would be nicer for this function to take a lambda that doesn't know about JS / WASM. Maybe something like `F: Fn(&str, &str) -> impl Future<std::result::Result<GetSymbolTableBuffers, String>>` (not sure if that's the right future type though). And then the caller of `get_symbolicate_response_impl` in `lib.rs` would create that lambda, and make it call the JS function and do the value and error conversion.

Reply:
Hmm, I guess with that modification, `symbolicate_response_assembler.rs` would still need to know about the `WasmMemBuffer` type because that's what's stored in the `GetSymbolTableBuffers` struct. Maybe we can improve that at some point by storing a `Box<>` of some trait object instead, with a trait that `WasmMemBuffer` implements. But we can leave that for later.